File: examples/javascript/constructs/pipe_operator.js
// Ported from examples/lumen/constructs/pipe_operator.lm by scripts/port_examples.py; edit the Lumen original, not this file.
function double(x) {
    return x * 2;
}

function add_one(x) {
    return x + 1;
}

function square(x) {
    return x * x;
}

console.log("Test: Pipe Operator");
console.log("Without pipe: square(add_one(double(5)))");
console.log(square(add_one(double(5))));
console.log("With pipe: 5 |> double() |> add_one() |> square()");
const result = square(add_one(double(5)));
console.log(result);
console.log("10 |> double():");
console.log(double(10));
function multiply(a, b) {
    return a * b;
}

console.log("3 |> double():");
const x = double(3);
console.log(multiply(x, 2));

File: examples/c/constructs/pipe_operator.c
// Ported from examples/lumen/constructs/pipe_operator.lm by scripts/port_examples.py; edit the Lumen original, not this file.
long double_(long x) {
    return x * 2;
}

long add_one(long x) {
    return x + 1;
}

long square(long x) {
    return x * x;
}

long multiply(long a, long b) {
    return a * b;
}

int main(void) {
    puts("Test: Pipe Operator");
    puts("Without pipe: square(add_one(double(5)))");
    printf("%ld\n", square(add_one(double_(5))));
    puts("With pipe: 5 |> double() |> add_one() |> square()");
    long result = square(add_one(double_(5)));
    printf("%ld\n", result);
    puts("10 |> double():");
    printf("%ld\n", double_(10));
    puts("3 |> double():");
    long x = double_(3);
    printf("%ld\n", multiply(x, 2));
    return 0;
}
